dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/router/state/AbstractStateRouter.java

```diff
@@ -125,7 +125,7 @@ public final BitList<Invoker<T>> route(
 
         routeResult = doRoute(invokers, url, invocation, needToPrintMessage, nodeHolder, messageHolder);
         if (routeResult != invokers) {
-            routeResult = invokers.and(routeResult);
+            routeResult = routeResult.and(invokers);
```

Comment:
Do not change it.

Reply:
In the new route rule, I will determine if the invoker object's memory address has changed to determine if it is a valid route. So this order change is necessary.

Reply:
replace with invokers.clone().and(routeResult)